### ISSUE TITLE
Impulse at line trace hit location

### ISSUE THREAD

**Issue by Konfyouzd** [jjenki11/MindSwapINC#1]
Okay... So remember how you said that we should use a projectile and I was so convinced that we didn't need the projectile?

Check out this solution... 

https://answers.unrealengine.com/questions/301427/how-to-add-force-or-impulse-at-the-hit-location-of.html

Despite being so sure that we didn't need it, I tried to spawn a projectile at some point yesterday and was completely unable to do it.  However, I was also completely unable to spawn an impulse there until late last night too so spawing the projectile has to be possible.  

I think we'll need to set up the projectile input parameters similar to how they're done in AAverageJoe::OnFire() in a blueprint.  OR... We can attempt to add the line trace code to the OnFire method and plug the resultant vectors from that into the SpawnProjectile (not sure if that's the actual name) function.

Seems like trying to spawn the projectile again is worth a shot.  They did almost exactly what you said.  Invisible projectile spawned at the impact location.  Maybe we can write something where the initial velocity of the projectile is smaller the farther away you are.  That might be easier to achieve than trying to muck with the impulse vectors.

**Comment by jjenki11** [comment]
well i hate to say i told you so but.... haha

i had used the old UE lots before this new gui and that is how it was done
previously which is why i suggested to do it like this, sorry if i dont
fully explain my reasoning and/or logic  - sometimes i just know its gotta
be done a certain way and can't explain why.

like you said we could probably have saved a lot of time last night messing
with that impulse vector, BUT it is awesome that we created a similar
effect via math.  i can guarantee you that certain functionality we will
try to implement later won't have a nifty little pipeline node to connect
and we will have to rely on our understanding the geometry of the scene to
create the effect.

good shit last night!  despite the deceptively simple task we tackled the
problem and gleaned a lot of insight into that new blueprint manager.
 thanks for being patient and willing to try things that defy logic lol
